app/src/main/java/com/google/android/soundchecker/utils/MultiChannelSineSource.kt

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.soundchecker.utils
+
+import android.media.MediaCodec
+
+/**
+ * Multi channel sine source where each channel has its own frequency and amplitude.
+ * When pulling from this source, the channels will be rendered in order and not combined.
+ */
+class MultiChannelSineSource : AudioSource() {
+
+    private var mSineSources = ArrayList<SineSource>()
+
+    fun addPartial(min: Float, current: Float, max: Float, enableSineSweep: Boolean = false) {
+        val sineSource = SineSource()
+        sineSource.getFrequencyPort().mMinimum = min
+        sineSource.getFrequencyPort().mMaximum = max
+        sineSource.getFrequencyPort().set(current)
+        if (enableSineSweep) {
+            sineSource.enableSineSweep()
+        }
+        mSineSources.add(sineSource)
+    }
+
+    fun getFrequencyPort(i: Int): ControlPort {
+        return mSineSources[i].getFrequencyPort()
```

Comment:
Is it needed to check if size is greater than i? Same question for other access to `mSinceSources`

Reply:
I prefer to fail fast for this as it's a catastrophic error if this happens

Reply:
Ack